tests/unit/utils/sendTasksUpdates.test.ts

```diff
@@ -24,11 +25,42 @@ describe("sendTaskUpdate function", () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
+  test("should throw an error if response status is not OK", async () => {
+    const url = config(mockEnv).TRACKING_CHANNEL_URL;
+    const formattedString =
+      `**${userName}** added an update to their task: [${taskTitle}](<${taskUrl}>)\n` +
+      `\n**Completed**\n${completed}\n\n` +
+      `**Planned**\n${planned}\n\n` +
+      `**Blockers**\n${blockers}`;
+    const bodyObj = {
+      content: formattedString,
+    };
+
+    jest
+      .spyOn(global, "fetch")
+      .mockResolvedValueOnce(
+        new JSONResponse("", { status: 400, statusText: "Bad Request" })
+      );
+
+    await expect(
+      sendTaskUpdate(
+        completed,
+        planned,
+        blockers,
+        userName,
+        taskId,
+        taskTitle,
+        mockEnv
+      )
+    ).rejects.toThrowError("Failed to send task update: 400 - Bad Request");
```

Comment:
Should we not check for the status code first?

Reply:
Thankyou for your suggestion, the changes has been made please check.

Reply:
As we are checking for **!response.ok** to throw the error this should work fine. Because our main objective for this test to send error when response.ok is false and even test coverage for this file is 100. So I guess we should not.

Reply:
but how can we be sure that it will always throw an status code of 400?